my_enumerables.rb

```diff
@@ -0,0 +1,190 @@
+    end
+    self
+  end
+
+  def my_each_with_index(pro = nil)
+    return to_enum unless block_given?
+
+    arr = to_a
+    i = 0
+
+    if pro
+      while i < arr.length
+        proc.cal arr[i], i
+        i += 1
+      end
+    else
+      while i < arr.length
+        yield arr[i], i
+        i += 1
+      end
+    end
+
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr = to_a
+    arr2 = []
+    arr.my_each_with_index do |_item, index|
+      arr2.push(arr[index]) if yield arr[index]
+    end
+    arr2
+  end
+
+  def my_all?(args = nil)
+    arr = to_a
+
+    arr.my_each_with_index do |item, index|
+      if block_given?
+        return false unless yield arr[index]
+      elsif args.is_a? Class
+        return false unless item.class.ancestors.include?(args)
+      elsif args.is_a? Regexp
+        return false unless item.to_s.match(args)
+      elsif args
+        return false if item != args
+      elsif item == false || item.nil?
+        return false
+      end
+    end
+
+    true
+  end
+
+  def my_any?(args = nil)
+    any = false
+    arr = to_a
+
+    if block_given?
+      my_each_with_index do |_item, index|
+        any = true if yield arr[index]
+      end
+    elsif args.is_a? Class
+      my_each_with_index do |item, _index|
+        any = true if item.class.ancestors.include?(args)
+      end
+    elsif args.is_a? Regexp
+      my_each_with_index do |item, _index|
+        any = true if item.match(args)
+      end
+    else
+      my_each_with_index do |item, _index|
+        any = true unless item == false || item.nil? || !item.to_s.match(args.to_s)
+      end
+    end
+    any
+  end
+
+  def my_none?(args = nil)
+    arr = to_a
+    any = true
+    arr.my_each_with_index do |item, index|
+      if block_given?
+        any = false if yield arr[index]
+      elsif args.is_a? Class
+        any = false if item.class.ancestors.include?(args)
+      elsif args.is_a? Regexp
+        any = false if item.match(args)
+      elsif args
+        any = false if args == item
+      else
+        any = false unless item == false || item.nil?
+      end
+    end
+    any
+  end
+
+  def my_count(arg = nil)
+    count = index = 0
+    arr = to_a
+    return arr.length if arg.nil? && !block_given?
+
+    while index < arr.length
+      if (arg.is_a?(Proc) && proc.call(arr[index])) ||
+         (arg == arr[index])
+        count += 1
+      elsif block_given?
+        count += 1 if yield arr[index]
+      end
+      index += 1
+    end
+    count
+  end
+
+  def my_map(pro = nil)
+    return to_enum unless block_given?
+
+    arr = to_a
+    arr2 = []
+    arr.my_each_with_index do |_item, index|
+      if block_given?
+        arr2.push(pro.call(arr[index])) if pro
+        arr2.push(yield(arr[index])) unless pro
+      elsif pro
+        arr2.push(pro.call(arr[index]))
+      else
+        arr2.push(yield arr[index])
+      end
+    end
+    arr2
+  end
+
+  def my_inject(arg = nil, sym = nil)
```

Comment:
This should combine all elements of enum by applying a binary operation specified by a block when an initial value is given.

Reply:
![Screenshot from 2021-01-08 12-02-30](https://user-images.githubusercontent.com/31193823/104036409-bbfa5480-51a9-11eb-9262-805be62e095e.png)